src/main/java/org/veupathdb/service/sr/response/Bases.java

```diff
@@ -12,9 +12,16 @@
 public class Bases {
   private static final Logger LOG = LogManager.getLogger(Bases.class);
 
-  public static byte[] getBasesForBedFeature(IndexedFastaSequenceFile sequences, BEDFeature feature){
+  public static byte[] getBasesForBedFeature(IndexedFastaSequenceFile sequences, BEDFeature feature) {
+    // Don't ask SAMTools for a feature whose end exceeds the length of the sequence.
+    long sequenceEnd = Math.min(sequences.getIndex().getIndexEntry(feature.getContig()).getSize(), feature.getEnd());
```

Comment:
The placement of this looks like a bug.  Do you mean to compare sequenceEnd on line 20?  If not, maybe move this to below the check.

Reply:
I actually think we should be using sequenceEnd on line 20. It's a pretty extreme edge case, but it could possibly matter.